oauth2_http/javatests/com/google/auth/oauth2/ServiceAccountCredentialsTest.java

```diff
@@ -241,6 +242,55 @@ public void getRequestMetadata_customTokenServer_hasAccessToken() throws IOExcep
   }
 
   @Test
+  public void refreshAccessToken_retries() throws IOException {
```

Comment:
This looks like multiple tests in one. Ideally each test should test a single piece of functionality. Please consider splitting it into multiple test cases. For example, for this particular big test:
- no retry needed case
- IOException case
- Http error code case
- non-retriable error case